core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java

```diff
@@ -81,20 +82,20 @@ private void importData(File file, int cancerStudyId) throws IOException, DaoExc
                 }
     
                 String sampleId = StableIdUtil.getSampleId(barCode);
-                String chrom = strs[1].trim(); 
+                String chrom = strs[headerIndexMap.get("chrom")].trim(); 
                 //validate in same way as GistitReader:
                 ValidationUtils.validateChromosome(chrom);
                 
-                long start = Double.valueOf(strs[2]).longValue();
-                long end = Double.valueOf(strs[3]).longValue();
+                long start = Double.valueOf(strs[headerIndexMap.get("loc.start")]).longValue();
+                long end = Double.valueOf(strs[headerIndexMap.get("loc.end")]).longValue();
                 if (start >= end) {
                     //workaround to skip with warning, according to https://github.com/cBioPortal/cbioportal/issues/839#issuecomment-203452415
                     ProgressMonitor.logWarning("Start position of segment is not lower than end position. Skipping this entry.");
                     entriesSkipped++;
                     continue;
                 }            
-                int numProbes = new BigDecimal((strs[4])).intValue();
-                double segMean = Double.parseDouble(strs[5]);
+                int numProbes = new BigDecimal((strs[headerIndexMap.get("num.mark")])).intValue();
+                double segMean = Double.parseDouble(strs[headerIndexMap.get("seg.mean")]);
```

Comment:
and these indices *will* be found because getIndexMap(line) will throw and exception if all required fields are not present.